File: 0950/solution.ts
function deckRevealedIncreasing(deck: number[]): number[] {
  deck.sort((a, b) => a - b);
  const res = Array.from({ length: deck.length }, () => 0);
  let i = 0;
  const queue: Array<number> = [];
  for (let i = 0; i < deck.length; i++) {
    queue.push(i);
  }
  while (queue.length) {
    const curr = queue.shift()!;
    res[curr] = deck[i++];
    if (queue.length) {
      queue.push(queue.shift()!);
    }
  }
  return res;
}

File: 0950/solution.py
from collections import deque


class Solution:
    def deckRevealedIncreasing(self, deck: list[int]) -> list[int]:
        deck.sort()
        D = len(deck)
        res = [None] * D
        queue = deque([i for i in range(D)])
        i = 0
        while queue:
            curr = queue.popleft()
            res[curr] = deck[i]
            i += 1
            if queue:
                queue.append(queue.popleft())
        return res
